core/schema/internalsdk.graphqls

```diff
@@ -0,0 +1,138 @@
+extend type Query {
+  """
+  Create a new function from the provided definition.
+  """
+  newFunction(def: FunctionDef!): Function!
```

Comment:
This is a bit different looking than our existing APIs, so worth a brief explanation.

I wanted to avoid having `with*` fields on the actual `Function` type since that closely mixes together APIs that anyone can call and the APIs that only an SDK should be calling internally. Only SDK internal implementations are allowed to create new `Function`s (it doesn't make sense to create a new one elsewhere, the engine wouldn't have anywhere to actually invoke it from).

So this is the only API for creating a new `Function`.

However, it's also worth noting that anyone is allowed to call `withFunction` on `Module`. This makes sense since it's fine to associate other pre-existing functions with a `Module` ([see this discord message about "monkeypatching](https://discord.com/channels/707636530424053791/1146498746432094208/1147281294112333844)")